File: WT_ASSIGNMENT-3/Bank-Account-Project/BankAccount.java
import java.util.Scanner;

class BankAccount {

    String accountNumber;
    String accountHolder;
    double balance;

    // Constructor
    BankAccount(String accountNumber, String accountHolder, double balance) {
        this.accountNumber = accountNumber;
        this.accountHolder = accountHolder;
        this.balance = balance;
    }

    // Deposit money
    void deposit(double amount) {
        if (amount > 0) {
            balance = balance + amount;
            System.out.println("Amount deposited successfully.");
        } else {
            System.out.println("Invalid amount.");
        }
    }

    // Withdraw money
    void withdraw(double amount) {
        if (amount > 0 && amount <= balance) {
            balance = balance - amount;
            System.out.println("Amount withdrawn successfully.");
        } else {
            System.out.println("Insufficient balance or invalid amount.");
        }
    }

    // Display balance
    void checkBalance() {
        System.out.println("Current Balance: Rs. " + balance);
    }

    // Display account details
    void displayDetails() {
        System.out.println("\n===== ACCOUNT DETAILS =====");
        System.out.println("Account Number: " + accountNumber);
        System.out.println("Account Holder: " + accountHolder);
        System.out.println("Balance: Rs. " + balance);
    }

    public static void main(String[] args) {

        Scanner sc = new Scanner(System.in);

        // Creating an account
        BankAccount account =
            new BankAccount("10245", "Rahul", 10000);

        int choice;

        do {
            System.out.println("\n===== BANK ACCOUNT SYSTEM =====");
            System.out.println("1. Deposit");
            System.out.println("2. Withdraw");
            System.out.println("3. Check Balance");
            System.out.println("4. Account Details");
            System.out.println("5. Exit");

            System.out.print("Enter your choice: ");
            choice = sc.nextInt();

            switch (choice) {

                case 1:
                    System.out.print("Enter deposit amount: ");
                    double depositAmount = sc.nextDouble();
                    account.deposit(depositAmount);
                    break;

                case 2:
                    System.out.print("Enter withdrawal amount: ");
                    double withdrawAmount = sc.nextDouble();
                    account.withdraw(withdrawAmount);
                    break;

                case 3:
                    account.checkBalance();
                    break;

                case 4:
                    account.displayDetails();
                    break;

                case 5:
                    System.out.println("Thank you for using the Bank Account System.");
                    break;

                default:
                    System.out.println("Invalid choice.");
            }

        } while (choice != 5);

        sc.close();
    }
}

File: WT_ASSIGNMENT-3/Bank-Account-Project/BankAccount.js
const readline = require("readline");

class BankAccount {

    constructor(accountNumber, accountHolder, balance) {
        this.accountNumber = accountNumber;
        this.accountHolder = accountHolder;
        this.balance = balance;
    }

    // Deposit money
    deposit(amount) {
        if (amount > 0) {
            this.balance = this.balance + amount;
            console.log("Amount deposited successfully.");
        } else {
            console.log("Invalid amount.");
        }
    }

    // Withdraw money
    withdraw(amount) {
        if (amount > 0 && amount <= this.balance) {
            this.balance = this.balance - amount;
            console.log("Amount withdrawn successfully.");
        } else {
            console.log("Insufficient balance or invalid amount.");
        }
    }

    // Display balance
    checkBalance() {
        console.log("Current Balance: Rs. " + this.balance);
    }

    // Display account details
    displayDetails() {
        console.log("\n===== ACCOUNT DETAILS =====");
        console.log("Account Number: " + this.accountNumber);
        console.log("Account Holder: " + this.accountHolder);
        console.log("Balance: Rs. " + this.balance);
    }
}


// Create readline interface
const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});


// Create bank account
const account = new BankAccount("10245", "Rahul", 10000);


// Function to display menu
function showMenu() {

    console.log("\n===== BANK ACCOUNT SYSTEM =====");
    console.log("1. Deposit");
    console.log("2. Withdraw");
    console.log("3. Check Balance");
    console.log("4. Account Details");
    console.log("5. Exit");

    rl.question("Enter your choice: ", function(choice) {

        switch (Number(choice)) {

            case 1:
                rl.question("Enter deposit amount: ", function(amount) {
                    account.deposit(Number(amount));
                    showMenu();
                });
                break;

            case 2:
                rl.question("Enter withdrawal amount: ", function(amount) {
                    account.withdraw(Number(amount));
                    showMenu();
                });
                break;

            case 3:
                account.checkBalance();
                showMenu();
                break;

            case 4:
                account.displayDetails();
                showMenu();
                break;

            case 5:
                console.log(
                    "Thank you for using the Bank Account System."
                );
                rl.close();
                break;

            default:
                console.log("Invalid choice.");
                showMenu();
        }
    });
}


// Start program
if (process.argv.includes("--demo")) {

    console.log("===== BANK ACCOUNT SYSTEM - DEMO =====");

    const demoAccount = new BankAccount("10245", "Rahul", 10000);

    console.log("\nInitial Balance:");
    demoAccount.checkBalance();

    console.log("\nDepositing Rs. 5000...");
    demoAccount.deposit(5000);

    console.log("\nWithdrawing Rs. 2500...");
    demoAccount.withdraw(2500);

    console.log("\nFinal Balance:");
    demoAccount.checkBalance();

    console.log("\nAccount Details:");
    demoAccount.displayDetails();

} else {

    showMenu();
}